### ISSUE TITLE
Panicked at Unknown Type

### ISSUE THREAD

**Issue by shellkr** [mmstick/systemd-manager#3]
I am using Arch and was planning to send it up to the AUR. It was no problem to build but when I try to run 'sudo systemd-manager-pkexec' I get this error:

    thread '<main>' panicked at 'Unknown Type: /usr/lib/systemd/system
    /org.freedesktop.systemd1.busname', src/systemd_dbus.rs:26

I am not using Gnome (Bspwm) but have all the related Gnome packages that is needed installed. Other Gnome apps works as they should. If I run 'systemd-manager-pkexec' It will ask me for my users password and then fail:

    ==== AUTHENTICATING FOR org.freedesktop.policykit.pkexec.systemd-manager ===
    Authentication is required to modify systemd services
    Authenticating as: myusername
    Password: 
    polkit-agent-helper-1: error response to PolicyKit daemon: 
    GDBus.Error:org.freedesktop.PolicyKit1.Error.Failed: No session for cookie
    ==== AUTHENTICATION FAILED ===
    Error executing command as another user: Not authorized

**Comment by mmstick** [comment]
`systemd-manager-pkexec` is just a shell script that launches the program, `systemd-manager`, with pkexec. The PolicyKit file needs to be in `/usr/share/polkit-1/actions/` to allow pkexec permissions to function. Although the error that you are receiving is because I don't have a match for `busname` types in my `UnitType` enum. I'll add that now.

### Code in question:

```
pub enum UnitType { Automount, Mount, Path, Scope, Service, Slice, Socket, Target, Timer }
impl UnitType {
	// Takes the pathname of the unit as input to determine what type of unit it is.
	pub fn new(pathname: &str) -> UnitType {
		match Path::new(pathname).extension().unwrap().to_str().unwrap() {
			"automount" => UnitType::Automount,
			"mount" => UnitType::Mount,
			"path" => UnitType::Path,
			"scope" => UnitType::Scope,
			"service" => UnitType::Service,
			"slice" => UnitType::Slice,
			"socket" => UnitType::Socket,
			"target" => UnitType::Target,
			"timer" => UnitType::Timer,
			_ => panic!("Unknown Type: {}", pathname),
		}
	}
}
```

**Comment by mmstick** [comment]
It should be fixed now on your system. (112b6f173c3584fc59e40fa62569baea752f186e)

**Comment by shellkr** [comment]
It seems to have fixed the first error but now if I run 'sudo systemd-manager-pkexec', 'systemd-manager-pkexec',  'sudo systemd-manager' and 'systemd-manager'. They all give the below error.

    thread '<main>' panicked at 'Unknown State: ['i', 'n', 'd', 'i', 'r', 'e', 'c', 't']', src/systemd_dbus.rs:44

**Comment by mmstick** [comment]
Same thing here: I don't have a matching enum for units with `indirect` states.

Although now I do: (c1c2fae2a8af01fdb731b3406323e319eb555d66)

**Comment by mmstick** [comment]
Pushed out another update supporting the rest of the states that I haven't implemented yet, apparently: (88843b5283c7d521e132ee52be0da592cdaa26cc)

**Comment by shellkr** [comment]
Great! Now it works and can close this ;)
